Library/src/Systems/SystemBase.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "CSP/Systems/SystemBase.h"
+
+#include "CSP/Multiplayer/EventBus.h"
+#include "Memory/Memory.h"
+
+namespace csp::systems
+{
+
+SystemBase::SystemBase() : WebClient(nullptr), EventBusPtr(nullptr)
+{
+}
+
+SystemBase::SystemBase(csp::web::WebClient* InWebClient, csp::multiplayer::EventBus* InEventBus) : WebClient(InWebClient), EventBusPtr(InEventBus)
+{
+	RegisterSystemCallback();
+}
+
+SystemBase::SystemBase(csp::multiplayer::EventBus* InEventBus) : WebClient(nullptr), EventBusPtr(InEventBus)
+{
+	RegisterSystemCallback();
+}
+
+SystemBase::~SystemBase()
+{
+	DeregisterSystemCallback();
+	EventBusPtr = nullptr;
+}
+
+void SystemBase::RegisterSystemCallback()
+{
+	if (!SystemCallback)
+	{
+		return;
+	}
+
+	if (EventBusPtr)
+		EventBusPtr->ListenEvent("DefaultEvent", this);
```

Comment:
It  may make more sense to make RegisterSystemCallback and DeregisterSystemCallback purely virtual because this DefaultEvent does nothing. However, this would mean that we have to implement events for all the systems that inherit from this class and are currently not using the event bus (about 10).
It may be more elegant to just have this function do nothing at all, as opposed to registering a default event.